pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -133,9 +133,17 @@ func (a Adapter) runBuildConfig(client *occlient.Client, parameters common.Build
 	if !isImageRegistryInternal {
 		secretName = regcredName
 	}
-	_, err = client.CreateDockerBuildConfigWithBinaryInput(commonObjectMeta, dockerfilePath, parameters.Tag, []corev1.EnvVar{}, buildOutput, secretName)
-	if err != nil {
-		return err
+
```

Comment:
Have you tested it with both internal and external registry?  In the case of internal registry, you always pass "" as secretName.